pkg/nsm/apis/common/common.proto

```diff
@@ -50,4 +54,20 @@ enum InterfacePreference {
     THIRD = 3;
     FORTH = 4;
     FIFTH = 5;
-}
\ No newline at end of file
+}
+
+message RemoteMechanism {
+    RemoteMechanismType type = 1;
+    map<string,string> constraints = 2;
```

Comment:
So... the naming of this field is an interesting problem.  I can agree that 'labels' is probably wrong.  Sometimes its 'constraints' and sometimes its 'parameters'... so not quite sure the right name for it.

Reply:
@edwarnicke parameters since it is more generic, I think could cover both constraints and parameters, but if come up with good name, I am perfectly ok to change it.